rskj-core/src/main/java/org/ethereum/core/BlockHeader.java

```diff
@@ -349,6 +353,10 @@ public Coin getMinimumGasPrice() {
         byte[] uncleCount = RLP.encodeBigInteger(BigInteger.valueOf(this.uncleCount));
         fieldToEncodeList.add(uncleCount);
 
+        if (this.ummRoot != null) {
```

Comment:
Should this be the condition an not `isActive(umm)`? For example, `getPendingState` always create a `null` `ummRoot`, even after activation... If coded this way, we will have two kind of blocks inside the node, some which encode to something with ummroots, and some which dont... is this intended? Sounds like it will complicate things at some point.